script/testScripts/gov/AddRewardsToNitroPool/ProposeAddRewardsToNitroPool.s.sol

```diff
@@ -0,0 +1,124 @@
+
+// Mock contract for NitroPool that will be used in the script. This is used to be very clear about what we are proposing.
+interface NitroPool {
+  function addRewards(uint256 amountToken1, uint256 amountToken2) external;
+}
+
+struct ProposalData {
+  address governanceAddress;
+  address nitroPoolAddress;
+  address rewardTokenAddress1;
+  address rewardTokenAddress2;
+  uint256 amountTokenReward1;
+  uint256 amountTokenReward2;
+}
+
+/// @title ProposeAddRewardsToNitroPool Script
+/// @author OpenDollar
+/// @notice Script to propose adding rewards to NitroPool via ODGovernance
+/// @dev NOTE This script requires the following env vars in the REQUIRED ENV VARS section below
+/// @dev This script is used to propose adding rewards to NitroPool, we first approve ERC20 to NitroPool and then call addRewards
+/// @dev The script will output a JSON file with the proposal data to be used by the QueueProposal and ExecuteProposal scripts
+/// @dev In the root, run: export FOUNDRY_PROFILE=governance && forge script --rpc-url <RPC_URL> script/testScripts/gov/AddRewardsToNitroPool/ProposeAddRewardsToNitroPool.s.sol
+contract ProposeAddRewardsToNitroPool is JSONScript {
+  function run() public {
+    /// REQUIRED ENV VARS ///
+
+    // using a struct so we don't get stack too deep errors
+    ProposalData memory data = ProposalData({
+      governanceAddress: vm.envAddress('GOVERNANCE_ADDRESS'),
+      nitroPoolAddress: vm.envAddress('NITRO_POOL_ADDRESS'),
+      rewardTokenAddress1: vm.envAddress('REWARD_TOKEN_ADDRESS_1'),
+      rewardTokenAddress2: vm.envAddress('REWARD_TOKEN_ADDRESS_2'),
+      amountTokenReward1: vm.envUint('AMOUNT_TOKEN_REWARD_1'),
+      amountTokenReward2: vm.envUint('AMOUNT_TOKEN_REWARD_2')
+    });
+
+    /// END REQUIRED ENV VARS ///
+
+    ODGovernor gov = ODGovernor(payable(data.governanceAddress));
+
+    // we need to support the case where we only have one reward token, so we need to dynamically build the targets and calldatas and avoid approving 0 amounts
+    // at most we will have 3 targets, 3 calldatas, value is always 0
+    bytes[] memory tempCalldatas = new bytes[](3);
+    address[] memory tempTargets = new address[](3);
+    // count is used to keep track of the number of targets and calldatas
+    uint256 count = 0;
+
+    if (data.amountTokenReward1 > 0) {
+      tempTargets[count] = data.rewardTokenAddress1;
+      tempCalldatas[count] =
+        abi.encodeWithSelector(IERC20.approve.selector, data.nitroPoolAddress, data.amountTokenReward1);
+      count++;
+    }
+
+    if (data.amountTokenReward2 > 0) {
+      tempTargets[count] = data.rewardTokenAddress2;
+      tempCalldatas[count] =
+        abi.encodeWithSelector(IERC20.approve.selector, data.nitroPoolAddress, data.amountTokenReward2);
+      count++;
+    }
+
+    tempTargets[count] = data.nitroPoolAddress;
+    tempCalldatas[count] =
+      abi.encodeWithSelector(NitroPool.addRewards.selector, data.amountTokenReward1, data.amountTokenReward2);
```

Comment:
Would this still work if amountTokenReward2 is 0?

Reply:
yes, the implementation of nitro pool take into consideration if amounts are `>0` 

```
    if (amountToken1 > 0) {
     ...
    }

    if (amountToken2 > 0) {
     ...
    }
```